src/shared/redirect.ts

```diff
@@ -0,0 +1,35 @@
+import { NextPageContext } from "next";
+import absolute from "next-absolute-url";
+import { getUserId } from "./session";
+
+export const redirect = async (ctx: NextPageContext) => {
```

Comment:
I think this is the old typing from getInitialProps and is missing a crucial piece! 
```suggestion
export const redirect: GetServerSideProps = async (ctx) => {
```